Tests/Serializer/BaseSerializerHandlerTest.php

```diff
@@ -0,0 +1,144 @@
+            ),
+            array(
+                'direction' => GraphNavigator::DIRECTION_SERIALIZATION,
+                'format' => 'xml',
+                'type' => 'foo',
+                'method' => 'serializeObjectToId',
+            ),
+            array(
+                'direction' => GraphNavigator::DIRECTION_DESERIALIZATION,
+                'format' => 'xml',
+                'type' => 'foo',
+                'method' => 'deserializeObjectFromId',
+            ),
+            array(
+                'direction' => GraphNavigator::DIRECTION_SERIALIZATION,
+                'format' => 'yml',
+                'type' => 'foo',
+                'method' => 'serializeObjectToId',
+            ),
+            array(
+                'direction' => GraphNavigator::DIRECTION_DESERIALIZATION,
+                'format' => 'yml',
+                'type' => 'foo',
+                'method' => 'deserializeObjectFromId',
+            ),
+        );
+
+        $methods = $serializer->getSubscribingMethods();
+
+        $this->assertSame($methods, $expectedMethods);
+    }
+
+    public function testSerializeObjectToIdWithDataIsInstanceOfManager()
+    {
+        $data = $this->getMock(
+            'Sonata\CoreBundle\Tests\Fixtures\Bundle\Serializer\FooSerializer',
+            array('getId'),
+            array(),
+            '',
+            false
+        );
+
+        $data->expects($this->once())
+            ->method('getId')
+            ->willReturn(1);
+
+        $manager = $this->getMock('Sonata\CoreBundle\Model\ManagerInterface');
+        $manager->expects($this->once())
+            ->method('getClass')
+            ->willReturn(get_class($data));
+
+        $context = $this->getMock('JMS\Serializer\Context');
+
+        $visitor = $this->getMock('JMS\Serializer\VisitorInterface');
+        $visitor->expects($this->once())
+            ->method('visitInteger')
+            ->with(1, array('foo'), $context)
+            ->willReturn(true);
+
+        $serializer = new FooSerializer($manager);
+
+        $this->assertTrue($serializer->serializeObjectToId($visitor, $data, array('foo'), $context));
+    }
+
+    public function testSerializeObjectToIdWithDataIsNotInstanceOfManager()
+    {
+        $data = $this->getMock(
+            'Sonata\CoreBundle\Tests\Fixtures\Bundle\Serializer\FooSerializer',
+            array(),
+            array(),
+            '',
+            false
+        );
+
+        $manager = $this->getMock('Sonata\CoreBundle\Model\ManagerInterface');
+        $manager->expects($this->once())
+            ->method('getClass')
+            ->willReturn('bar');
+
+        $context = $this->getMock('JMS\Serializer\Context');
+
+        $visitor = $this->getMock('JMS\Serializer\VisitorInterface');
+        $visitor->expects($this->never())
+            ->method('visitInteger');
+
+        $serializer = new FooSerializer($manager);
+
+        $serializer->serializeObjectToId($visitor, $data, array('foo'), $context);
```

Comment:
What is the repetition exactly I didn't get it. The mocks are a bit different then the mock in previous test method.